flink-external-resources/flink-external-resource-gpu/src/test/java/org/apache/flink/externalresource/gpu/GPUDriverTest.java

```diff
@@ -0,0 +1,121 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.externalresource.gpu;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.util.FlinkException;
+import org.apache.flink.util.OperatingSystem;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.Set;
+
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assume.assumeTrue;
+
+/**
+ * Tests for {@link GPUDriver}.
+ */
+public class GPUDriverTest extends TestLogger {
+
+	private static final String TESTING_DISCOVERY_SCRIPT_PATH = "src/test/resources/testing-gpu-discovery.sh";
+	private static final String TEST_SCRIPT_PATH = "src/test/resources/test-privilege-allocate.sh";
+
+	@Test
+	public void testGPUDriverWithTestScript() throws Exception {
+		final int gpuAmount = 2;
+		final Configuration config = new Configuration();
+		config.setString(GPUDriver.DISCOVERY_SCRIPT_PATH, TESTING_DISCOVERY_SCRIPT_PATH);
+
+		final GPUDriver gpuDriver = new GPUDriver(config);
+		final Set<GPUInfo> gpuResource = gpuDriver.retrieveResourceInfo(gpuAmount);
+
+		assertThat(gpuResource.size(), is(gpuAmount));
+	}
+
+	@Test
+	public void testPrivilegeMode() throws Exception {
+		assumeTrue(OperatingSystem.isLinux());
+		testExistWithNonZero("test_privilege_mode");
+	}
+
+	@Test
+	public void testCheckDead() throws Exception {
+		assumeTrue(OperatingSystem.isLinux());
+		testExistWithNonZero("test_check_dead");
+	}
+
+	@Test
+	public void testAssignFile() throws Exception {
+		assumeTrue(OperatingSystem.isLinux());
+		testExistWithNonZero("test_assign_file");
+	}
+
+	private void testExistWithNonZero(String cmd) throws Exception {
+		final ProcessBuilder processBuilder = new ProcessBuilder(TEST_SCRIPT_PATH, cmd);
+		processBuilder.redirectErrorStream(true);
+		final Process process = processBuilder.start();
+		final int exitValue = process.waitFor();
+		if (exitValue != 0) {
+			final BufferedReader stdoutReader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+			final String stdout = stdoutReader.lines().collect(StringBuilder::new, StringBuilder::append, StringBuilder::append).toString();
+			log.error("Script exit with non-zero {}, output: {}", exitValue, stdout);
+			throw new Exception("Script exist with non-zero.");
+		}
+	}
```

Comment:
These should belong to another file, with the name `GpuDiscoveryScriptTest` or something similar.
These cases are not testing the behavior of `GPUDriver`, but the behavior of `gpu-discovery-util.sh`.